pkg/resources/user.go

```diff
@@ -181,61 +254,210 @@ func ReadUser(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	u, err := snowflake.ScanUserDescription(rows)
-	if err != nil {
-		return err
-	}
-	if err = d.Set("name", u.Name.String); err != nil {
-		return err
+	if user.Name != nil {
```

Comment:
similar comment could applied here.

Maybe we should rearrange both of these and introduce something like a really simple builder which we could use like (pseudocode):
```
b
  .set("name", user.Name)
  .set("comment", user.Comment)
  .set("login_name", user.LoginName)
  .build() // this last part would try to apply all above and return first error which we can check against OR something like that
```

\* and this is just an idea, which we can discuss and implement it in following PRs if accepted

Reply:
We could do that if we care about short-circuiting(i.e. stopping after first encountered error).
Another option would be to use `errors.Join` so we would have sth like
```
err := errors.Join(
d.set("name", user.Name) 
...
)

```
I wonder if aggregating errors like that isn't a better option. Those actions are independent and we probably want to give as much information to the user as possible, if we discover multiple errors, we should report all of them.

I'm not modifying the code just yet, we should discuss it further

Reply:
i dont like use a builder pattern for this kind of problem. In the new plugin framework, which we will eventually be migrating to, they use models to read from and save the state. See https://developer.hashicorp.com/terraform/plugin/framework/resources/create.

You might have a ThingResource with a model like:
```
// ThingResourceAPIModel describes the API data model.
type ThingResourceAPIModel struct {
    Name string `json:"name"`
    Id   string `json:"id"`
}
```
which then can be marshalled / unmarshalled using helper functions:

```
  // Read Terraform plan data into the model
    resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
```

and

```
    // Save data into Terraform state
    resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)

```

Reply:
Ok, I guess I'll keep it for now then